### PR TITLE
Using map reduce and filter for Packets

### PR DESCRIPTION
**Pull request by eddiemoore**

### CODE REVIEW

File: src/packets/packet.js [hunk 1]
```diff
@@ -22,36 +20,34 @@ export default class Packet {
...
+          PacketFields[this.type].hasOwnProperty(key) &&
+          fields[key] !== null &&
+          typeof fields[key] !== 'undefined'
+        ) {
+          return { [key]: fields[key] }
+        }
+        return null
+      })
+      .filter(i => i !== null)
+      .reduce((l, c) => Object.assign({}, l, c), {})
```
Comment (bblurock) [inline src/packets/packet.js]: I don't know why put an empty object at the end? :sweat_smile:

Comment (eddiemoore) [inline src/packets/packet.js]: It is for the initial value for the reduce

File: src/packets/packet.js [hunk 2]
```diff
@@ -61,13 +57,10 @@ export default class Packet {
...
-      json.F = {}
-
-      Object.keys(this.fields).forEach(key => {
-        if (PacketFields[this.type].hasOwnProperty(key)) {
-          json.F[PacketFields[this.type][key]] = this.fields[key]
-        }
-      })
+      json.F = Object.keys(this.fields)
+        .map(key => ({ [PacketFields[this.type][key]]: this.fields[key] }))
+        .filter(i => i !== null)
```
Comment (bblurock) [inline src/packets/packet.js]: should this check with `undefined`? if nothing being return in map, result shall be 'undefined' I suppose.

File: src/packets/packet.js (hunk 1 above)
Comment (bblurock) [inline src/packets/packet.js]: ah, that is part of reduce, my bad, on lined code make me think that is part of arrow function XD

File: src/packets/packet.js (hunk 2 above)
Comment (eddiemoore) [inline src/packets/packet.js]: Actually I realise this line is not needed. Cause it will never be null after the map.

**Comment by bblurock** [comment]
Looks good, 👍

**Pull request merged**